tests/lib/tools/fs-state

```diff
@@ -0,0 +1,115 @@
+#!/bin/bash
+
+show_help() {
+    echo "usage: fs-state wait-for-file [filename] [iterations] [sleep-time]"
+    echo "usage: fs-state mock-dir [direname]"
+    echo "usage: fs-state mock-dir-backup-real [direname]"
+    echo "usage: fs-state restore-dir [direname]"
+    echo "usage: fs-state mock-file [filename]"
+    echo "usage: fs-state mock-file-backup-real [filename]"
+    echo "usage: fs-state restore-file [filename]"
+}
+
+wait_for_file() {
+    local file="$1"
+    local iters="$2"
+    local sleep_time="$3"
+
+    for _ in $(seq "$iters"); do
+        if [ -e "$file" ]; then
+            echo "File exists: $file"
+            return 0
+        fi
+        sleep "$sleep_time"
+    done
+    echo "File does not exists: $file"
+    return 1
+}
+
+mock_dir() {
+    local dir="$1"
+    if ! [ -d "$dir" ]; then
+        echo "Creating dir: $dir"
+        mkdir -p "$dir"
+        touch "$dir.fake"
+    fi
+}
+
+mock_dir_backup_real() {
+    local dir="$1"
+    if [ -d "$dir" ]; then
+        mv "$dir" "$dir.back"
+    fi
+    echo "Creating dir: $dir"
+    mkdir -p "$dir"
+    touch "$dir.fake"
+}
+
+restore_dir() {
+    local dir="$1"
+
+    echo "Cleaning dir: $dir"
+    if [ -f "$dir.fake" ]; then
```

Comment:
Should restore fail if mock was not called?

Reply:
same answser